src/Operations.jl

```diff
@@ -758,7 +759,17 @@ function install_archive(
 
         # Move content to version path
         !isdir(version_path) && mkpath(version_path)
-        mv(unpacked, version_path; force=true)
+        unpacked_tree_hash = GitTools.tree_hash(unpacked)
+        try # protect against interrupted moves
+            mv(unpacked, version_path; force=true)
```

Comment:
I think the actually correct thing to do here is to instead change line 733 to:

```
dir = mktempdir(dirname(version_path))
```

Then change this `mv()` to a `rename()`.  I believe this should be guaranteed atomic, because we can safely assume that users have not created separate filesystems to hold specific versions of packages.  The only caveat to be aware of here I believe is that `rename()` can fail if `version_path` exists, which could happen if, for instance, an older Julia without this pidlocking is installing the same thing at the same time.  So I guess we just have to handle that exception, check that the directory exists, then consider that a success.  If you want to be paranoid, you could check the treehash in such an event.

Reply:
Looking through the Artifacts codepath, I think you basically want to call `_mv_temp_artifact_dir(unpacked, version_path)` here.

Reply:
Sounds good. `_mv_temp_artifact_dir` internally set permissions, which we do after `install_archive` has returned, so I just disabled that here.